packages/babel-runtime/test/unittests/wrapRegExp.test.js

```diff
@@ -0,0 +1,16 @@
+import wrapRegExp from "../../helpers/esm/wrapRegExp.js";
+
+describe("wrapRegExp", () => {
+  it("should handle maliciously crafted substitutions", () => {
+    const pattern = "(foo)";
+    const groups = { group: 1 };
+    const myRegExp = wrapRegExp(pattern, groups);
+    const targetStr = "foofoo";
+    const str = "$<".repeat(1e5) + "group";
+    const startTime = Date.now();
+    myRegExp[Symbol.replace](targetStr, str);
+    // This test will fail when 1000ms is passed
+    const timeTaken = Date.now() - startTime;
+    expect(timeTaken).toBeLessThan(1000);
```

Comment:
```suggestion
    const str = "$<".repeat(1e5) + "group>";
    const startTime = Date.now();
    const result = myRegExp[Symbol.replace](targetStr, str);
    
    // This test will fail when 1000ms is passed
    const timeTaken = Date.now() - startTime;
    expect(timeTaken).toBeLessThan(1000);
    
    expect(result).toBe("foo");
```

Note: there is currently a bug on `main` that causes `result` to be `$undefinedfoo` instead of `foo`.

I didn't try this PR, but if it causes the result to be `$<$<$<$<$<....$undefinedfoo` or `$<$<$<$<$<$<...foo`, it's a regression.

Reply:
I suggested this as an edit, but it should be actually be a separate test (because the `>` after `group` in str makes the performance problem disappear even without this patch).

Reply:
> $undefinedfoo

Yes, we didn't check whether the group name actually exist or if it is a valid regexp identifier.